### PR TITLE
OCPBUGS-27844: Fix task for attaching IPv6 subnet to router.

### PR DESCRIPTION
**Pull request by gryf**

### CODE REVIEW

File: upi/openstack/network.yaml [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 - hosts: all
   gather_facts: no
+  vars:
+    _interfaces: ["{{ os_subnet }}"]
```
Comment (MaysaMacedo) [inline upi/openstack/network.yaml:11]: how about we name this `router_interfaces` instead, so it becomes more descriptive?

Comment (MaysaMacedo) [reply upi/openstack/network.yaml:11]: also can you try converting the string to list with `{{ os_subnet | list }}` instead?

Comment (gryf) [reply upi/openstack/network.yaml:11]: that would produce `['n', 'a', 'm', 'e']` i the os_subnet would hold `name` value.

**Review by mandre — COMMENTED** [review]
Looks good but I first want to see this exercised in CI before merging.
Depends on https://github.com/openshift/release/pull/47872/ and an upcoming patch to add this as a pre-submit to the installer repo.

File: upi/openstack/network.yaml [hunk 2]
```diff
@@ -32,24 +34,18 @@
...
+      router_interfaces: " {{ router_interfaces + [ os_subnet6 ] }} "
+    when: os_subnet6 is defined
+
   - name: 'Create external router'
     openstack.cloud.router:
       name: "{{ os_router }}"
       network: "{{ os_external_network }}"
-      interfaces:
-      - "{{ os_subnet }}"
+      interfaces: "{{ router_interfaces }}"
```
Comment (mandre) [inline upi/openstack/network.yaml:46]: The job [currently fails with](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_installer/8087/pull-ci-openshift-installer-master-e2e-openstack-dualstack-upi/1765189720091922432/artifacts/e2e-openstack-dualstack-upi/upi-install-openstack/build-log.txt):
```
TASK [Create external router] **************************************************
fatal: [localhost]: FAILED! => {"changed": false, "extra_data": {"data": null, "details": null, "response": "None"}, "msg": "No Subnet found for  ['ocp-subnet-v4'"}
```

I suspect we need to remove the quotes:
```suggestion
      interfaces: {{ router_interfaces }}
```

Comment (gryf) [reply upi/openstack/network.yaml:46]: Well, by removing quotas will make ansible-playbook to complain about wrong syntax:
```console
Syntax Error while loading YAML.
  found unacceptable key (unhashable type: 'AnsibleMapping')
....
The offending line appears to be:

      network: "{{ os_external_network }}"
      interfaces: {{ router_interfaces }}
                   ^ here
...
```
no matter if I initialize `router_interfaces` with `"{{ [os_subnet] }}"` or `["{{ os_subnet }}"]`

Comment (gryf) [reply upi/openstack/network.yaml:46]: Although I've seen this error before (i.e. taking part of the list as a string `['ocp-subnet-v4'`). Let me dig a little bit more.

Comment (MaysaMacedo) [reply upi/openstack/network.yaml:46]: @gryf another option is to just create a separate task that will create the router with two interfaces

**Comment by gryf** [comment]
Okay, it seems workaround with two different tasks worked.

**Comment by MaysaMacedo** [comment]
@gryf given NFV job is not required, so we can skip the restest
@mandre can you take another look on this PR?

**Pull request merged**